### PR TITLE
Removes broken feature and handles right click

### PR DESCRIPTION
**Pull request by pols12**
I believe usual left click did not lead to l.facebook.com tracker anymore (Facebook probably uses another way to track clicks).

This refactoring handles right click preventing all Facebook events to be triggered by mouse-clicking on external links (outside facebook.com and facebookwww.onion domains).

You’ll find details in commit messages.

Please feel free to cherry pick any commit, they are all in [CC0](https://creativecommons.org/publicdomain/zero/1.0/deed) license.

### CODE REVIEW

**Comment by Mte90** [comment]
Hi pols12 thanks for the contribution!
I am just checking that code and I am trying to understand the difference with the actual one.

As now we get all the anchors and do a check if it is from a specific domain and if it include some specific links to do some exclusion and also clean the url.

Your code instead takes just the link that include lm.facebook and clean the url.

I will do some tests anyway but just to be sure, because facebook change the url on click so the actual behaviour was built to be sure to act always.

**Comment by pols12** [comment]
In fact, I have removed most of current code because I did not understand it:
- sometimes it seems to be defective (some foreach expect to work for every cases)
- sometimes, I don’t understand why you want to clean the link (all internal facebook.com links → Facebook can track you anyway server-side)

In this PR, I check l.facebook links, but I am pretty sure this is actually useless because all links seem to be loaded as fully external links (so http://google.com and not http://l.facebook.com/l.php?u=http://google.com)

**To summarize**: `<a>` tag are inserted by Facebook without tracking, Facebook uses events to add them a tracking system. So this PR proposes to block all events which could be triggered on <a> tag.
I keep a cleanup function because I’m not sure whether extension’s event are triggered before Facebook’s event.

**Comment by Mte90** [comment]
Thanks for the clarification!

I tried to block all the facebook events that change the links but with the old and new design with React in some cases they wasn't detected and the code evolved to support those cases.

I will test it in the next days and I will let you know :-)

**Comment by Mte90** [comment]
I tested and everything works but `fbclid` is nor cleaned by the url. I am checking the code.
